src/java/org/apache/cassandra/net/async/RebufferingByteBufDataInputPlus.java

```diff
@@ -249,4 +253,50 @@ public ByteBufAllocator getAllocator()
     {
         return channelConfig.getAllocator();
     }
+
+    /**
+     * Consumes bytes in the stream until the given length
+     *
+     * @param writer
+     * @param len
+     * @return
+     * @throws IOException
+     */
+    public long consumeUntil(BufferedDataOutputStreamPlus writer, long len) throws IOException
+    {
+        long copied = 0; // number of bytes copied
+        while (copied < len)
+        {
+            int position = buffer.position();
+            int remaining = buffer.remaining();
+            if (remaining == 0)
+            {
+                try
+                {
+                    reBuffer();
+                } catch (EOFException e)
+                {
+                    throw new EOFException("EOF after " + copied + " bytes out of " + len);
+                }
+                position = buffer.position();
+                remaining = buffer.remaining();
+                if (remaining == 0)
```

Comment:
The codebase that I've been through doesn't use braces for single statement blocks. I assumed that was the convention. I can change it in my PR to follow the Sun style but we should ideally audit and change the code style in the future to avoid confusion.

Reply:
I think we need to raise this on the mailing list to possibly modify the code style, because for a long time it has been standard practice for many contributors (myself included) to elide the braces for single line statements, because otherwise the code becomes illegible.

Reply:
Even you've done it!  :)

https://github.com/aweisberg/cassandra/commit/1491a40b7b4ea2723bcf22d870ee514b47ea901b#diff-baae9b53db48ff1ed346e469f7506545R192

Reply:
Style guide says

“Always include braces for nested levels of conditionals and loops. Only avoid braces for single level.”

Skipping for single level has always been ok.

Reply:
My bad. I missed the last part of it.

And me doing something is not an endorsement of it being a good idea/right!